src/components/Switch.jsx

```diff
@@ -26,17 +26,24 @@ module.exports = React.createClass({
   },
 
   getToggle(open) {
-    let { parent } = this.props
+    if (open) return null
 
-    return !open ? (
-      <SwitchToggle onClick={ this._onToggle } secondary={ parent } />
-    ) : null
+    return (<SwitchToggle ref="toggle"
+                          onClick={ this._onToggle }
+                          secondary={ this.props.parent } />)
   },
 
   getNav(open, blockTypes) {
+    if (!open) return null
+
     let { app, parent, position } = this.props
 
-    return open ? React.createElement(SwitchNav, { app, blockTypes, parent, position }) : null
+    return (<SwitchNav ref="nav"
+                       app={ app }
+                       blockTypes={ blockTypes }
+                       onExit={ this._onNavExit }
+                       parent={ parent }
+                       position={ position } />)
```

Comment:
Making a note to myself to revisit `<SwitchNav />`. It knows too much, I'd like to just sent it an `onClick` handler or something and not have to pass down `app`, `parent,` or `position`